packages/myst-to-jats/src/backmatter.ts

```diff
@@ -112,7 +113,7 @@ export function citeToJatsRef(key: string, data: CitationJson): Element {
   return {
     type: 'element',
     name: 'ref',
-    attributes: { id: key },
+    attributes: { id: `${key}${notebookArticleSuffix(state)}${slugSuffix(state)}` },
```

Comment:
The article suffix (`-article`) comes in on the article representation of the notebook; slug suffix is only used on citations so there is not collision on the reference ids across different backmatters.